jupyterhub/apihandlers/base.py

```diff
@@ -18,9 +18,12 @@ class APIHandler(BaseHandler):
     def content_security_policy(self):
         return '; '.join([super().content_security_policy, "default-src 'none'"])
 
+    def get_content_type(self):
+        return 'application/json'
+
     def set_default_headers(self):
         super().set_default_headers()
-        self.set_header('Content-Type', 'application/json')
+        self.set_header('Content-Type', self.get_content_type())
```

Comment:
Great!

Though, is this the right place for this? I think this might be the default behaviour of tornado's `set_default_headers()`, where it calls `self.get_content_type()`? If this is overwritten in `BaseHandler`, this seems like behaviour that is nice enough that `BaseHandler` could reimplement this instead of only applying to the `APIHandler` base class? 

For more context on where I'm coming from, If you look at the change I made in [this problem](https://github.com/jupyter/notebook/pull/3368/files#diff-7a881c97595330d2dab8c7ef400b7372L24), I had to make that change because setting the header directly didn't work — instead of `'text/x-yaml'` it was returning `'application/octet-stream'`.

Reply:
>  I think this might be the default behaviour of tornado's set_default_headers()

That's what I thought, too, but it's only the default behavior for StaticFileHandler and its subclasses. So I copied the tornado implementation from StaticFileHandler to our set_default_headers.

> this seems like behaviour that is nice enough that BaseHandler could reimplement this

Good call. I'll move it up.